packages/hydrogen/src/csp/csp.ts

```diff
@@ -0,0 +1,47 @@
+import {useContext} from 'react';
+import cspBuilder from 'content-security-policy-builder';
+import {HydrogenServerContext} from '../HydrogenServerProvider';
+
+export const useNonce = () => {
+  return useContext(HydrogenServerContext).nonce;
+};
+
+export function createCSPHeader(
+  nonce?: string,
+  directives: Record<string, string[] | string | boolean> = {},
```

Comment:
@blittle I guess `createCSPHeader(undefined, { ... })` will be rare enough that we don't need to change the signature?
Or should we go directly for `createCSPHeader({ nonce, ... })`?